File: Jack.Business/Roupa.vb
Public Class Roupa

    Implements ICrud(Of Model.Roupa, Integer)

    Public Sub New()

    End Sub

    Public Function Delete(oTipo As Model.Roupa) As Boolean Implements ICrud(Of Model.Roupa, Integer).Delete
        Dim oDados As Data.Roupa = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Roupa
            blnRetorno = oDados.Delete(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno

    End Function

    Public Function Find(Identifier As Integer) As Model.Roupa Implements ICrud(Of Model.Roupa, Integer).Find

        Dim oDados As Data.Roupa = Nothing
        Dim lstRetorno As IList(Of Model.Roupa) = Nothing

        Try
            oDados = New Data.Roupa
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Insert(oTipo As Model.Roupa) As Boolean Implements ICrud(Of Model.Roupa, Integer).Insert
        Dim oDados As Data.Roupa = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Roupa
            blnRetorno = oDados.Insert(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function

    Public Function LoadAll() As IList(Of Model.Roupa) Implements ICrud(Of Model.Roupa, Integer).LoadAll
        Dim oDados As Data.Roupa = Nothing
        Dim lstRetorno As IList(Of Model.Roupa) = Nothing

        Try
            oDados = New Data.Roupa
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Update(oTipo As Model.Roupa) As Boolean Implements ICrud(Of Model.Roupa, Integer).Update
        Dim oDados As Data.Roupa = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Roupa
            blnRetorno = oDados.Update(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function
End Class

File: Jack.Application/Roupa.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Jack.Application
{
    public class Roupa : ICrud<Model.Roupa, int>
    {

        public Roupa()
        {

        }

        public bool Delete(Model.Roupa oTipo)
        {
            Repository.RepRoupa oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Repository.RepRoupa();
                blnRetorno = oDados.Delete(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;

        }

        public Model.Roupa Find(int Identifier)
        {

            Repository.RepRoupa oDados = null;
            Model.Roupa oRetorno = null;

            try
            {
                oDados = new Repository.RepRoupa();
                oRetorno = oDados.Find(Identifier);
            }
            catch (Exception ex)
            {
                oRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return oRetorno;

        }

        public bool Insert(Model.Roupa oTipo)
        {
            Repository.RepRoupa oDados = null;
            bool blnRetorno = false;


            try
            {
                oDados = new Repository.RepRoupa();
                blnRetorno = oDados.Insert(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }

        public IList<Model.Roupa> LoadAll()
        {
            Repository.RepRoupa oDados = null;
            IList<Model.Roupa> lstRetorno = null;

            try
            {
                oDados = new Repository.RepRoupa();
                lstRetorno = oDados.LoadAll();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public bool Update(Model.Roupa oTipo)
        {
            Repository.RepRoupa oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Repository.RepRoupa();
                blnRetorno = oDados.Update(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }
    }
}
